chef/cookbooks/glance/recipes/ceph.rb

```diff
@@ -24,20 +27,22 @@
 else
   # If external Ceph cluster will be used,
   # we need install ceph client packages
-  if node[:platform] == "suse" && File.exists?("/etc/ceph/ceph.client.admin.keyring")
-    package "ceph-common"
-    package "python-ceph"
-  else 
+  if File.exists?(ceph_conf) && File.exists?(admin_keyring)
+    if node[:platform] == "suse"
+      package "ceph-common"
+      package "python-ceph"
+    end
+  else
     return
```

Comment:
I apologize for suggesting this code because I now see it can be improved:

```ruby
return unless File.exists?(ceph_conf) && File.exists?(admin_keyring)

if node[:platform] == "suse"
  package "ceph-common"
  package "python-ceph"
end
```